R/specify.R

```diff
@@ -148,35 +148,50 @@ parse_variables <- function(x, formula, response = NULL,
     attr(x, "explanatory") <- f_rhs(formula)
   }
   
-  if (is_nuat(x, "response")) {
+  # Check response and explanatory variables to be appropriate for later use
+  if (!has_response(x)) {
     stop_glue("Supply not `NULL` response variable.")
   }
-  
-  if (!(as.character(attr(x, "response")) %in% names(x))) {
-    stop_glue(
-      'The response variable `{attr(x, "response")}` cannot be found in this ',
-      'dataframe.'
-    )
-  }
+  check_var_correct(x, "response")
+  check_var_correct(x, "explanatory")
   
   # If there's an explanatory var
-  if (has_explanatory(x)) {
-    if (!as.character(attr(x, "explanatory")) %in% names(x)) {
+  check_vars_different(x)
+  
+  x
+}
+
+check_var_correct <- function(x, var_name) {
+  var <- attr(x, var_name)
+  
+  # Variable (if present) should be a symbolic column name
+  if (!is.null(var)) {
+    if (!rlang::is_symbolic(var)) {
+      stop_glue("The {var_name} variable should be symbolic.")
```

Comment:
I expect very few infer users will know what the term "symbolic" means.  Perhaps: "The {var_name} should be symbolic (not in quotation marks)."

It's possible that they'd put in something that's neither a symbol or a character string, but I'd expect this would cover 95% of the mistakes.

Reply:
Then, how about "The {var_name} should be a bare variable name (not a string in quotation marks)."?

Reply:
That's great